Client/Frontend/Browser/LoginsHelper.swift

```diff
@@ -285,20 +284,25 @@ class LoginsHelper: TabContentScript {
                 return loginData.toDict(formSubmitURL: formSubmitURL)
             }
             
-            // The frame must belong to the same security origin
-            if let currentHost = currentHost,
-               !currentHost.isEmpty,
-               currentHost == frameHost {
-                // Prevent XSS on non main frame
-                // If it is not the main frame, return username only, but no password!
-                // Chromium does the same on iOS.
-                // Firefox does NOT support third-party frames or iFrames.
+            // Check for current tab has a url to begin with
+            // and the frame is not modified
+            guard let currentURL = tab?.webView?.url,
+                  LoginsHelper.checkIfFrameInfoNotModified(
+                    url: currentURL,
+                    frameScheme: securityOrigin.protocol,
+                    frameHost: securityOrigin.host,
+                    framePort: securityOrigin.port) else {
+                return nil
+            }
+            
+            // Prevent XSS on non main frame
```

Comment:
This comment should move to inside the `checkIfFrameInfoNotModified` where it does the `frameHost` comparison.